protos/flyteidl/admin/common.proto

```diff
@@ -280,6 +280,14 @@ message Annotations {
     map<string, string> values = 1;
 }
 
+// Environment variable values to be applied to an execution resource.
+// In the future a mode (e.g. OVERRIDE, APPEND, etc) can be defined
+// to specify how to merge environment variables defined at registration and execution time.
+message Envs {
+    // Map of custom environment variables to be applied to the execution resource.
+    map<string, string> values = 1;
```

Comment:
I don't have a good reason but we don't need to look up, we just need uniqueness. I think this should be a list.. i know it's a dict in python but it's a list in k8s...
We just have had too many "arrrrrgh, I wish we made it a list" moments in the past (variables in lps come to mind where maintaining original ordering is desired)